avro-fastserde/src/main/java/com/linkedin/avro/fastserde/PrimitiveFloatList.java

```diff
@@ -25,23 +31,33 @@
  * - It re-implements {@link #compareTo(GenericArray)}, {@link #equals(Object)} and {@link #hashCode()}
  *   in order to leverage the primitive types, rather than causing unintended boxing.
  *
- * TODO: Provide arrays for other primitive types.
+ *   Using ByteBuffer to speed up float-array deserialization: We allocate ByteBuffer to store the raw bytes from
+ *   BinaryDecoder and deserialize them only during array element access. We cache the results into the elements array
+ *   after the first get access of the array so that sub-sequent array access are fast.
+ *
+ *   TODO: Provide arrays for other primitive types.
  */
 public class PrimitiveFloatList extends AbstractList<Float>
```

Comment:
Instead of changing this class, could we change `PrimitiveFloatList` to an interface, make the existing class one of the implementations, and then create additional implementations? I think this would be cleaner, and would allow us to benchmark the various implementations against one another in a nicer way. It would also make it very clear for our users which methods are intended to be public as part of the interface.